### PR TITLE
Update mqtt pipeline to support all required mqtt packet types

### PR DESCRIPTION
**Pull request by suchen-sci**

### CODE REVIEW

File: pkg/object/mqttproxy/client.go [hunk 1]
```diff
@@ -257,9 +244,35 @@ func (c *Client) processPuback(puback *packets.PubackPacket) {
...
+			if ctx.Disconnect() {
+				c.close()
+				return errors.New("pipeline set disconnect")
+			}
+			if ctx.Drop() {
+				return errors.New("pipeline set drop")
+			}
+		}
+	}
+	return nil
```
Comment (localvar) [inline pkg/object/mqttproxy/client.go:264]: return early to avoid nesting.

Comment (suchen-sci) [reply pkg/object/mqttproxy/client.go]: updated!

File: pkg/object/pipeline/mock.go [hunk 2]
```diff
@@ -87,60 +91,70 @@ func (m *MockMQTTFilter) Init(filterSpec *FilterSpec) {
...
-}
-
-func (m *MockMQTTFilter) clientDisconnect() map[string]struct{} {
-	m.mu.Lock()
-	defer m.mu.Unlock()
-	ans := make(map[string]struct{})
-	for k, v := range m.disconnect {
-		ans[k] = v
+	disconnect := make(map[string]struct{})
+	for k, _ := range m.disconnect {
```
Comment (zhao-kun) [inline pkg/object/pipeline/mock.go:138]: Eliminating the lint's complains

**Review by localvar — APPROVED** [review]

File: pkg/object/mqttproxy/client.go [hunk 3]
```diff
@@ -257,9 +244,40 @@ func (c *Client) processPuback(puback *packets.PubackPacket) {
...
+
 func (c *Client) processSubscribe(packet *packets.SubscribePacket) {
 	logger.SpanDebugf(nil, "client %s subscribe %v with qos %v", c.info.cid, packet.Topics, packet.Qoss)
-	err := c.broker.topicMgr.subscribe(packet.Topics, packet.Qoss, c.info.cid)
+	err := c.runPipeline(packet)
+	if err != nil {
+		logger.SpanDebugf(nil, "client %v process subscribe %v failed, %v", c.info.cid, packet.Topics, err)
+		return
+	}
+
```
Comment (zhao-kun) [inline pkg/object/mqttproxy/client.go:279]: Duplicated codes in each packet routines, according to DRY, it's very great if you refactor it”

Comment (suchen-sci) [reply pkg/object/mqttproxy/client.go:279]: ... Not all packets run pipeline... if we received some packets, it's an error, like `Connect` about connect established, `Connack`, `Suback`, `Unsuback`; some packet we do nothing to it, like `Pingresp`; some packet we just send response like `Pingreq`... Is this want you mean?

Comment (suchen-sci) [reply pkg/object/mqttproxy/client.go]: updated!

**Review by zhao-kun — APPROVED** [review]

**Review by samutamm — APPROVED** [review]

**Pull request merged**